ui/src/app/kvm/views/KVMDetails/KVMDetailsHeader/KVMDetailsHeader.tsx

```diff
@@ -3,34 +3,46 @@ import { useEffect } from "react";
 import { Spinner } from "@canonical/react-components";
 import { usePrevious } from "@canonical/react-components/dist/hooks";
 import { useDispatch, useSelector } from "react-redux";
-import { useParams } from "react-router";
 import { Link, useLocation } from "react-router-dom";
 
 import type { KVMSelectedAction, KVMSetSelectedAction } from "../KVMDetails";
 
 import SectionHeader from "app/base/components/SectionHeader";
-import type { RouteParams } from "app/base/types";
 import KVMActionFormWrapper from "app/kvm/components/KVMActionFormWrapper";
 import PodDetailsActionMenu from "app/kvm/components/PodDetailsActionMenu";
+import { getActionTitle as getPodActionTitle } from "app/kvm/utils";
+import { getActionTitle as getMachineActionTitle } from "app/machines/utils";
 import { actions as podActions } from "app/store/pod";
 import podSelectors from "app/store/pod/selectors";
+import type { Pod } from "app/store/pod/types";
 import { PodType } from "app/store/pod/types";
 import type { RootState } from "app/store/root/types";
 
 type Props = {
+  id: Pod["id"];
   selectedAction: KVMSelectedAction | null;
   setSelectedAction: KVMSetSelectedAction;
 };
 
+const getActionTitle = (selectedAction: KVMSelectedAction) => {
+  // This is a reliable of differentiating a machine action from a pod action,
+  // but we should eventually try to have a consistent shape between them.
+  // https://github.com/canonical-web-and-design/maas-ui/issues/3017
+  if (typeof selectedAction === "object" && "name" in selectedAction) {
```

Comment:
`selectedAction` can be null so this needs to be:
`if (selectedAction && typeof selectedAction === "object" && "name" in selectedAction) {`